pkg/webhook/server_test.go

```diff
@@ -0,0 +1,148 @@
+/*
```

Comment:
can you add a test case to test the case where the server exceeds cert refresh interval? you can change `defaultCertRefreshInterval` to a smaller interview. I am worried that if you start a server on the same port immediately after shutdown, you will have port conflict.

Reply:
IIUC 
https://github.com/kubernetes-sigs/controller-runtime/pull/260/files#diff-8b0412ec4fd52af1419bd0fcd0f2e101R121 is doing what you ask. The server restarts multiple times and rotates the certs.